packages/plugin-legacy/src/index.ts

```diff
@@ -204,15 +204,13 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
           config.build.cssTarget = 'chrome61'
         }
 
-        if (genLegacy) {
```

Comment:
I'm not sure why this was here in the first place. There might be implications in removing this that I might be missing as I'm not extremely familiar with the plugin's code base.

Reply:
This was here because we only wanted to modify the `config.build.target` when we generate both legacy and modern chunks. We shouldn't set `config.build.target` always here as by default it'll make it have a lower version by default, and breaking.

Reply:
After thinking hard about this because I was still confused, I realized that I probably had a fundamental misunderstanding. I tried to control the target browsers purely through the @vitejs/plugin-legacy configuration and did not realize that Vite itself has this `config.build.target` options that can/should be set in the `vite.config.js`. Maybe in part due to the warning

```
`plugin-legacy overrode 'build.target'. You should pass 'targets' as an option to this plugin with the list of legacy browsers to support instead.
```

that you get when activating legacy chunks. It sounds like you shouldn't set `build.target` at all when using plugin-legacy. But when not generating legacy chunks, but still desiring certain syntax transform, you need to set `modernTargets` *and* `build.target` to have the proper syntax transform in the modern chunks. I suppose that the documentation and/or warnings could potentially be improved. I might think about how exactly, but that would be something for a separate PR.

Long story short, the condition here should stay because when generating legacy /and/ modern chunks. The syntax transforms for `build.target`/`targets` only need to be performed for the legacy chunks because the modern chunks are used only with the `modernTargets` browsers. It is also important to note (if I understand correctly) that the `config.build.target` is effectively only used for the modern chunks and the legacy chunks make an additional pass through Babel with `targets` set in the Babel configuration for the syntax transforms.